### PR TITLE
created stack logic

### PR DESCRIPTION
**Pull request by AndriiBardin**

### CODE REVIEW

File: src/main/java/core/stack/MateStack.java [hunk 1]
```diff
@@ -1,19 +1,46 @@
...
 
+import java.util.EmptyStackException;
+
 public class MateStack<T> {
-    public void push(T value) {
+    private Node<T> lastNode;
+    private int size;
 
+    public void push(T value) {
+        lastNode = new Node<> (lastNode,value);
```
Comment (tbondare) [inline src/main/java/core/stack/MateStack.java]: пробела между входными данными нет

**Review by tbondare — CHANGES_REQUESTED** [review]

**Review by KseniiaMakarova — COMMENTED** [review]
please write your solution in a new branch - check out `git reset --soft` command for example, it will help to remove the commit from master branch

**Review by tbondare — APPROVED** [review]

File: src/main/java/core/stack/MateStack.java [hunk 2]
```diff
@@ -1,19 +1,46 @@
...
+
+    private void sizeCheck() {
+        if (size == 0) {
+            throw new EmptyStackException();
+        }
+    }
+
+    private static class Node<T> {
+        T value;
+        Node<T> previous;
```
Comment (KseniiaMakarova) [inline src/main/java/core/stack/MateStack.java:39]: access modifiers

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]
the status was not pushed to the table previously

**Pull request closed without merging**